t/plugin/xml-json-conversion.t

```diff
@@ -0,0 +1,222 @@
+no_long_string();
+no_root_location();
+no_shuffle();
+log_level('info');
+run_tests;
+
+__DATA__
+
+=== TEST 1: sanity
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.xml-json-conversion")
+            local conf = {from = "xml", to = "json"}
+
+            local ok, err = plugin.check_schema(conf)
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+done
+--- no_error_log
+[error]
+
+
+
+=== TEST 2: create a route with the plugin which set from=xml and to=json
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "xml-json-conversion": {
+                            "from": "xml",
+                            "to": "json"
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    },
+                    "uri": "/hello"
+                }]]
+            )
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+=== TEST 3: test for unsupported operation
+--- request
+GET /hello
+<people>
+ <person>
+   <name>Manoel</name>
+   <city>Palmas-TO</city>
+ </person>
+</people>
+--- more_headers
+Content-Type: text/html
+--- error_code: 400
+--- response_body
+{"message":"Operation not supported"}
+--- no_error_log
+[error]
+
+=== TEST 4: test for unsupported operation
+--- request
+GET /hello
+<people>
+ <person>
+   <name>Manoel</name>
+   <city>Palmas-TO</city>
+ </person>
+</people>
+--- more_headers
+Content-Type: application/json
+--- error_code: 400
+--- response_body
+{"message":"Operation not supported"}
+--- no_error_log
+[error]
+
+
+=== TEST 5: verify in argument
+--- config
+    location /t {
+        content_by_lua_block {
+            local headers = {}
+            headers["Content-Type"] = "text/xml"
+            local t = require("lib.test_admin").test
+            local code, body = t('/hello',
+                ngx.HTTP_GET,
+                [[<people>
+                     <person>
+                       <name>Manoel</name>
+                       <city>Palmas-TO</city>
+                     </person>
+                   </people>]],
+               nil,
+               headers
+            )
+
+            if code > 200 then
+                ngx.status = code
+                ngx.say(err)
+                return
+            end
+
+            ngx.say(body)
```

Comment:
should check the request body is converted to json format?

Reply:
done.